cirq-core/cirq/ops/common_gates.py

```diff
@@ -92,6 +92,30 @@ class XPowGate(eigen_gate.EigenGate):
     def __init__(
         self, *, exponent: value.TParamVal = 1.0, global_shift: float = 0.0, dimension: int = 2
     ):
+        """Initialize an XPowGate.
+
+        Args:
+            exponent: The t in gate**t. Determines how much the eigenvalues of
+                the gate are scaled by. For example, eigenvectors phased by -1
+                when `gate**1` is applied will gain a relative phase of
+                e^{i pi exponent} when `gate**exponent` is applied (relative to
+                eigenvectors unaffected by `gate**1`).
+            global_shift: Offsets the eigenvalues of the gate at exponent=1.
+                In effect, this controls a global phase factor on the gate's
+                unitary matrix. The factor is:
+
+                    exp(i * pi * global_shift * exponent)
+
+                For example, `cirq.X**t` uses a `global_shift` of 0 but
+                `cirq.rx(t)` uses a `global_shift` of -0.5, which is why
+                `cirq.unitary(cirq.rx(pi))` equals -iX instead of X.
```

Comment:
Consider updating the formula on line 75 above to account for global shift. Then you can refer to whatever letter you choose to use for global shift in the formula from the description of the argument here. This is what you effectively do on line 98 for exponent=t.

The benefit of this is that it allows the reader to unambiguously identify the gate they will get for given arguments.

Reply:
Done, assuming I understood you correctly. PTAL

Reply:
Yes, that's exactly it! :-)

I haven't checked that the math is correct. Might be worth calling `np.linalg.eig(cirq.unitary(...))` on one of the gates with a few values of exponent and global_shift and check that the eigenvalues are correct.

Reply:
(To be clear: I think we do this in unit tests, but doing it manually would give you an opportunity for compare the output against the math in the comment.)